pyomo/contrib/solver/common/base.py

```diff
@@ -29,17 +28,41 @@
 from pyomo.core.base import SymbolMap
 from pyomo.core.base.label import NumericLabeler
 from pyomo.core.staleflag import StaleFlagManager
-from pyomo.contrib.solver.config import SolverConfig, PersistentSolverConfig
-from pyomo.contrib.solver.util import get_objective
-from pyomo.contrib.solver.results import (
+from pyomo.scripting.solve_config import default_config_block
+from pyomo.contrib.solver.common.config import SolverConfig, PersistentSolverConfig
+from pyomo.contrib.solver.common.util import get_objective
+from pyomo.contrib.solver.common.results import (
     Results,
     legacy_solver_status_map,
     legacy_termination_condition_map,
     legacy_solution_status_map,
 )
 
 
-class SolverBase(abc.ABC):
+class Availability(IntEnum):
+    """
+    Class to capture different statuses in which a solver can exist in
+    order to record its availability for use.
+    """
+
+    FullLicense = 2
+    LimitedLicense = 1
+    NotFound = 0
+    BadVersion = -1
+    BadLicense = -2
+    NeedsCompiledExtension = -3
+
+    def __bool__(self):
+        return self._value_ > 0
+
+    def __format__(self, format_spec):
+        return format(self.name, format_spec)
+
+    def __str__(self):
+        return self.name
```

Comment:
When I saw this, I thought, "I'm surprised we need this". Then I looked at the version that was removed from within `SolverBase` and saw that we dropped some useful comments. Why not port those comments? Also, I still find it odd that we need to override `__str__` and `__format__`. Should we use a different type of enum?

Reply:
Oh, we need `IntEnum` for `__bool__`...